spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -473,5 +473,30 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
    batchSize: Option[Int] = None): Array[(ValidationResult, ValidationMethod[T])] = {
     Evaluator(this).test(dataset, vMethods, batchSize)
   }
+
+  /**
+   * Connect the current module to the current Graph with start nodes and end nodes
+   * @param startEnd: current start nodes and end nodes, they are both empty or both not empty
+   * @return
+   */
+  def toGraphNodes(startEnd: (Array[ModuleNode[T]], Array[ModuleNode[T]])):
```

Comment:
rename it to 'toGraph'?

Why use a tuple as input instead of two parameters?

def toGraph(starts: Array[ModuleNode[T]]): Array[ModuleNode[T]]

and 

def toGraph(): Array[ModuleNode[T]]

These methods take inputs node and output the end nodes. The second one creates some Identity nodes then invoke the first one.

Reply:
ok, got it

Reply:
I don't think your implementation is align with my suggestion